### PR TITLE
feat: revamped taxonomy sort script

### PR DESCRIPTION
**Pull request by alexgarel**
to really sort everything (within an entry)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -240,8 +240,10 @@ front_build:
 
 
 checks: front_build front_lint check_perltidy check_perl_fast check_critic
+# TODO: add check_taxonomies when taxonomies ready
 
 lint: lint_perltidy
+# TODO: add lint_taxonomies when taxonomies ready
```
Comment (benbenben2) [inline Makefile:246]: Is it dependent on this: https://github.com/openfoodfacts/openfoodfacts-server/pull/9843 ?

Comment (alexgarel) [reply Makefile:247]: Not really, but the other one is now merged.

It's more when taxonomy will be sorted !

File: scripts/taxonomies/sort_each_taxonomy_entry.pl [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	}
+	if ($b =~ /^(\w+):(\w+)$/) {
+		$b_prefix = $1;
+		$b = $2;
+	}
+	if ($a_prefix && $b_prefix) {
+		return $a_prefix cmp $b_prefix if ($a_prefix ne $b_prefix);
+	}
+	return 0 if ($a eq $b);
+	# en takes precedence over all others
```
[the remark this answers is not in the document]
Comment (alexgarel) [reply scripts/taxonomies/sort_each_taxonomy_entry.pl]: fixed, I now never change first entry (for now). I think it's a different decision to change the main language !

**Review by stephanegigandet — APPROVED** [review]
Looks good, tested on a few files like units and nutrients

**Pull request merged**